src/dotnet-interactive/Connection/PackageAcquisition.cs

```diff
@@ -0,0 +1,30 @@
+﻿// Copyright (c) .NET Foundation and contributors. All rights reserved.
+// Licensed under the MIT license. See LICENSE file in the project root for full license information.
+
+using System.Reflection;
+using Microsoft.DotNet.Interactive.Formatting;
+
+namespace Microsoft.DotNet.Interactive.App.Connection;
+
+internal static class PackageAcquisition
+{
+    internal static string InferCompatiblePackageVersion()
+    {
+        var informationalVersion = typeof(Formatter)
+                                   .Assembly
+                                   .GetCustomAttribute<AssemblyInformationalVersionAttribute>()
+                                   .InformationalVersion;
+
+        if (informationalVersion.Contains("+"))
+        {
+            informationalVersion = informationalVersion.Split("+")[0];
+        }
+
+        var splitInformationalVersion = informationalVersion.Split('.');
+
+        var version = splitInformationalVersion.Length is 4
+                          ? string.Join(".", splitInformationalVersion[0..4])
+                          : informationalVersion;
+        return version;
```

Comment:
Are all the conditions (i.e. versions that contain `+`, versions that contain more than 4 parts etc.) in this function reachable in practice for the versions we ship? It may be a good idea to include comments with an example or two for the (supported) formats we expect to encounter.

It would also be nice to have tests that exercise these cases - but I am guessing this may not be easy and perhaps it will not be worth it once #2871 is addressed fully...

Reply:
The number of variations is endless and in practice the build configuration decides what will actually be here. I've already seen a few variants that I didn't expect but that we don't care about, like `1.0.0.ci-12345.1`. I need to see this running with official bits before deciding the heuristic is acceptable.